docs/TerraToolkitUpgradeGuide-v3.0.0.md

```diff
@@ -0,0 +1,121 @@
+
+Terra toolkit now provides a default webpack config. This config will need to be referenced by the app level webpack config to add in an entry (unless consuming terra-dev-site). We recommend using [webpack-merge](https://github.com/survivejs/webpack-merge) to combine configs.
+The default webpack config is now a function that will flex between production and development modes by passing in the -p flag when compiling with webpack. For more about webpack config as a function go [here](https://webpack.js.org/configuration/configuration-types/).
+If you are consuming this webpack config to create another webpack config, don't forget that you need to execute the function first.
+
+```javascript
+const defaultWebpackFunction = require('terra-toolkit/lib/webpack/webpack.config');
+
+const defaultWebpackConfig = defaultWebpackFunction(); // This will give you the default dev webpack config.
+
+defaultWebpackConfig.entry = {
+    'index': path.resolve(path.join(__dirname, '..', 'Index')),
+  };
+
+module.exports = defaultWebpackConfig;
+
+```
+
+Translation Aggregation has been added to the default webpack config. By default all of the terra supported locales are included. If you need to customize this list, see the I18N aggregation section below. To completely disable translation aggregation within the webpack build to add your own, you can pass the environment variable `--env.disableAggregateTranslations` to the webpack command.
+
+```bash
+webpack --config src/webpack/webpack.config --env.disableAggregateTranslations
+```
+
+The `webpack` command is available to applications consuming terra-toolkit.
+
+The major change with webpack4 plugins is that the mini-css-extract-plugin now replaces extract-text-webpack-plugin which is no longer maintained.
+
+## I18N aggregation
+In a previous release, the aggregation translations script (`tt:aggregate-translations`) was added to terra-toolkit. In this release a new configuration option has been added. You can specify the aggregate-translations configuration through a `terraI18n.config.js` file in your root project directory.
+
+### Example
+```javascript
+const fse = require('fs-extra');
+const i18nConfig = {
+  baseDir: process.cwd(),
+  directories: [./translations],
+  fileSystem: fse,
+  locales: ['ar', 'en', 'en-US', 'en-GB', 'es', 'es-US', 'es-ES', 'de', 'fi-FI', 'fr', 'fr-FR', 'pt', 'pt-BR'],
+};
+
+exports.config = i18nConfig;
+```
+
+## Serve
+Terra-toolkit now provides a couple of scripts to help serve your site for development, testing, or as a doc site.
+
+### serve
+Serve is a replacement for webpack-dev-server. Behind the scenes it's using [webpack-serve](https://github.com/webpack-contrib/webpack-serve).
+Serve is a hot reloading server and does not work on IE 10 or below. See [compatible browsers](https://caniuse.com/#feat=websockets). Use serve-static for IE 10 testing.
```

Comment:
I'm outsourcing examples to the Readme in the serve folder in the script. For local exploratory testing people might want a hot reloading server. For any wdio tests, it will use serve-static by default.